### PR TITLE
Added 3 algorithms in cpp and java.

### PR DESCRIPTION
**Pull request by SobhanBera**
**Hi!** 
------

I have added bubble sort, the sieve of Eratosthenes in CPP and java, linear search in CPP and java and I will keep contributing to this project in future too.
Thanks

### CODE REVIEW

**Comment by Danushka96** [comment]
Hi @SobhanBera 
Please follow the Contribution Guidelines which can be found [here ](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md)carefully, 
Thanks

**Comment by RavinduSachintha** [comment]
> ## **Hi!**
> I have added bubble sort, the sieve of Eratosthenes in CPP and java, linear search in CPP and java and I will keep contributing to this project in future too.
> Thanks

Hi @SobhanBera , you need to be assigned for an issue before creating a PR. we have already assigned someone else to BubbleSort in CPP issue. Go through the current open issue list and create a new one without conflict.
Thank you. :+1:

File: algorithms/ar-bsrh/cpp/bubble_sort.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+using namespace std;
+
+void swap(int *i, int *j) {
+	int temp = *i;
+	*i = *j;
+	*j = temp;
+}
+
+void bubbleSort(int arr[], int n) {
+	bool swapped;
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/cpp/bubble_sort.cpp:11]: Codacy found an issue: [The scope of the variable 'swapped' can be reduced.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6278826)

File: algorithms/ar-bsrh/java/LinearSearch.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import java.util.*;
+
+public class LinearSearch {
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/java/LinearSearch.java:3]: Codacy found an issue: [The utility class name 'LinearSearch' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper|Constants)'](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6278826)

File: algorithms/ar-bsrh/java/LinearSearch.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    public static void main (String[] args) {
+        Scanner sc = new Scanner(System.in);
+        int n, x, q;
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/java/LinearSearch.java:6]: Codacy found an issue: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6278826)

File: algorithms/ar-bsrh/java/SieveOfEratosthenes.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import java.util.*;
+
+public class SieveOfEratosthenes {
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/java/SieveOfEratosthenes.java:3]: Codacy found an issue: [The utility class name 'SieveOfEratosthenes' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper|Constants)'](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6278826)

File: algorithms/ar-bsrh/java/SieveOfEratosthenes.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	// Avalue in prime[i] will be false if ith is Not a prime.
+	//providing output as final prime number upto n.
+	public static void sieveOfEratosthenes(int n) {
+	    boolean primes[] = new boolean[n+1];
+		for(int i = 0; i < n + 1; ++i) {
+			primes[i] = true;
+		}
+
+	    for (int p=2; p*p<=n; p++) {
+		    if (primes[p] == true) {
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/java/SieveOfEratosthenes.java:24]: Codacy found an issue: [Avoid unnecessary comparisons in boolean expressions](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6278826)

File: algorithms/ar-bsrh/cpp/bubble_sort.cpp (hunk 1 above)
Comment (SobhanBera) [reply algorithms/ar-bsrh/cpp/bubble_sort.cpp:11]: Thanks for your help
Please assign me too with any algorithm I will try my best to contribute.
I am not too familiar with the issue section.
Thanks

**Comment by RavinduSachintha** [comment]
@SobhanBera We cannot accept this PR as a valid one so far. Because,

- Before making a PR you should be assigned for an issue. You can create one. We will assign you to that.
- Each PR should only be linked to a single issue and vice versa.
- Each PR should contain file changes related to a single issue and an issue should be related to a single algorithm or data-structure.

If you wish to contribute to this repository, we are very happy to help you. Otherwise, we will have to close this PR without merging.

**Comment by Danushka96** [comment]
Closed due to inactivity

**Pull request closed without merging**